src/linux_ip_socket/linux_ip_socket.cpp

```diff
@@ -0,0 +1,257 @@
+/**@file
+ * This file is part of the TASTE Linux Runtime.
+ *
+ * @copyright 2021 N7 Space Sp. z o.o.
+ *
+ * TASTE Linux Runtime was developed under a programme of,
+ * and funded by, the European Space Agency (the "ESA").
+ *
+ * Licensed under the ESA Public License (ESA-PL) Permissive,
+ * Version 2.3 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://essr.esa.int/license/list
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "linux_ip_socket.h"
+
+#include <cstdint>
+#include <cstdlib>
+#include <cstring>
+#include <iostream>
+#include <string>
+
+#include <arpa/inet.h>
+#include <netdb.h>
+#include <netinet/in.h>
+#include <sys/socket.h>
+#include <sys/types.h>
+
+#include <drivers_config.h>
+
+extern "C"
+{
+#include <Broker.h>
+}
+
+linux_ip_socket_private_data::linux_ip_socket_private_data()
+    : m_thread(1, 65536)
+{
+}
+
+namespace taste {
+
+void
+LinuxIpSocketPoll(void* private_data)
```

Comment:
Maybe split this to some functions as it's a little bit overwhelming

Reply:
Done